plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/maven2/Maven2ProxyRecipe.groovy

```diff
@@ -0,0 +1,152 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2008-2015 Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.repository.maven.internal.maven2
+
+import javax.annotation.Nonnull
+import javax.inject.Inject
+import javax.inject.Named
+import javax.inject.Provider
+import javax.inject.Singleton
+
+import org.sonatype.nexus.repository.Format
+import org.sonatype.nexus.repository.RecipeSupport
+import org.sonatype.nexus.repository.Repository
+import org.sonatype.nexus.repository.Type
+import org.sonatype.nexus.repository.httpclient.HttpClientFacet
+import org.sonatype.nexus.repository.maven.internal.HostedHandler
+import org.sonatype.nexus.repository.maven.internal.MavenArtifactMatcher
+import org.sonatype.nexus.repository.maven.internal.MavenFacetImpl
+import org.sonatype.nexus.repository.maven.internal.MavenHeadersHandler
+import org.sonatype.nexus.repository.maven.internal.MavenMetadataMatcher
+import org.sonatype.nexus.repository.maven.internal.MavenProxyFacet
+import org.sonatype.nexus.repository.maven.internal.VersionPolicyHandler
+import org.sonatype.nexus.repository.negativecache.NegativeCacheFacet
+import org.sonatype.nexus.repository.negativecache.NegativeCacheHandler
+import org.sonatype.nexus.repository.proxy.ProxyHandler
+import org.sonatype.nexus.repository.search.SearchFacet
+import org.sonatype.nexus.repository.security.SecurityHandler
+import org.sonatype.nexus.repository.storage.StorageFacetImpl
+import org.sonatype.nexus.repository.types.HostedType
+import org.sonatype.nexus.repository.types.ProxyType
+import org.sonatype.nexus.repository.view.ConfigurableViewFacet
+import org.sonatype.nexus.repository.view.Route
+import org.sonatype.nexus.repository.view.Router
+import org.sonatype.nexus.repository.view.ViewFacet
+import org.sonatype.nexus.repository.view.handlers.TimingHandler
+
+import static org.sonatype.nexus.repository.http.HttpHandlers.notFound
+
+/**
+ * Maven 2 proxy repository recipe.
+ *
+ * @since 3.0
+ */
+@Named(Maven2ProxyRecipe.NAME)
+@Singleton
+class Maven2ProxyRecipe
+    extends RecipeSupport
+{
+  static final String NAME = "maven2-proxy"
```

Comment:
minor but IMO for groovy we should prefer single quotes for constants and double quotes for interpolated strings